source/baseObject.py

```diff
@@ -283,6 +282,12 @@ def getScript(self,gesture):
 				return self._gestureMap[identifier].__get__(self, self.__class__)
 			except KeyError:
 				continue
+			except AttributeError:
+				log.exception((
+					"Did you miss to initialize the base class?"
+					f"\nMRO={self.__class__.__mro__}"

```

Comment:
I haven't tested this out, so happy to go with your recommendation but in other situations I've found `__qualname__` really helpful. Particularly in cases where we have matching class names in different modules because the package, module, and class name are specified.
For anyone unfamiliar see https://stackoverflow.com/a/58110139/346188

Reply:
@feerrenrut wrote:
> in other situations I've found `__qualname__` really helpful.

I agree, thanks for pointing it out. I indeed recommended to use it when we were working on enhancing the log emitted by threads.

By the way, do you prefer your suggested change from https://github.com/nvaccess/nvda/pull/11166#discussion_r425635010 or @CyrilleB79's version from https://github.com/nvaccess/nvda/pull/11166#discussion_r425668376?

Reply:
@feerrenrut, I've just tested the current approach with a nested class and it appears __module__ and __qualname__ are already output as part of the class representation:
```
ERROR - baseObject.ScriptableObject.getScript (14:09:23.711) - winInputHook (5740):
Did you miss to initialize the base class?
MRO=(<class 'globalPlugins.i5446.GlobalPlugin'>, <class 'globalPlugins.i5446.Outer.Inner'>, <class 'globalPluginHandler.GlobalPlugin'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'object'>)
```

Thus, I guess no change is necessary here.
Still, please let me know your preference regarding the phrasing.